File: vca-backend-spring-boot-gp/src/main/java/com/vca/app/services/SegmentServiceImpl.java
package com.vca.app.services;

import com.vca.app.entity.Segment;
import com.vca.app.repositories.SegmentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class SegmentServiceImpl implements SegmentService {

	@Autowired
	SegmentRepository segmentRepository;

	@Override
	public List<Segment> getAllSegments() {
		return segmentRepository.findAll();
	}
}

File: vca-backend-dot-net-gp-Final/Services/Segments/SegmentServiceImpl.cs
using VCA.Models;
using VCA.Repositories;
namespace VCA.Services.Segments
{
    public class SegmentServiceImpl : ISegmentService
    {
        private readonly AppDbContext _context;// DI

        public SegmentServiceImpl(AppDbContext context)
        {
            _context = context;
        }


        public List<Segment> GetAllSegments()
        {
            return _context.Segments.ToList();
        }

    }
}
